grpclog/loggerv2_test.go

```diff
@@ -61,3 +66,264 @@ func checkLogForSeverity(s int, b []byte) error {
 	}
 	return nil
 }
+
+// makeLogFuncErrr returns a logger printFunc which generates a test error,
+// for testing that the log func is not called
+func makeLogFuncErr(t *testing.T) func() string {
+	return func() string {
+		t.Errorf("want Logger func to not be called for io.Discard")
+		return ""
+	}
+}
+
+// logFuncStr is a string used via testLogFuncOk and
+// testCheckLogContainsFuncStr to test the logger printFunc.
+const logFuncStr = "called-func"
+
+// logFuncOk is a logger printFunc to write a constant string
+func logFuncOk() string {
+	return logFuncStr
+}
+
+// checkLogContainsFuncStr checks that the logger buffer logBuf contains the
+// string written by logFuncOk
+func checkLogContainsFuncStr(t *testing.T, logBuf []byte) {
+	if !bytes.Contains(logBuf, []byte(logFuncStr)) {
+		t.Errorf("want logger func to be called and print '%v', got '%v'", logFuncStr, string(logBuf))
+	}
+}
+
+// checkWrite checks that the log buffer buf was written as expected.
+// The discard argument indicates there should not have been a write to the
+// buffer.
+// The logType indicates the type of the log, only used for better error
+// messages.
+// The msg is the string expected to have been written. Note we check
+// "contains," since the log will have a prefix.
+// The isJSON arg inddicates whether the logger was configured to write JSON.
+func checkWrite(t *testing.T, buf *bytes.Buffer, discard bool, logType string, msg string, isJSON bool) {
+	bts, err := buf.ReadBytes('\n')
+	if discard {
+		if err == nil {
+			t.Fatalf("want discard %v to not write, got '%v'", logType, string(bts))
```

Comment:
for comparison style guide recommends mentioning got before want https://google.github.io/styleguide/go/decisions#got-before-want

Reply:
Please change to this pattern everywhere else as well

Reply:
Changed